### PR TITLE
Initial Structure and Action

### PR DESCRIPTION
**Pull request by carsongee**
Here I've laid down the initial structure of the project for design discussion and changes.

It is laid out as a series of actions that have backing library modules to carry out those functions with a command line interface on top (for now).  Eventually I intend to have entry points via django URLs and using the django-rest-framework to back those, but for now, the command line API will work just as well.

I've also laid down a document project with the design document included and tried to make everything as clear as possible there since this is pretty big diff.  Docs at http://orcoursetrion.readthedocs.org/en/feature-initial/

For now the only action is to create a new empty repository in the studio org. I didn't want to get to far down the implementation path (i.e. teams, hooks, etc), before deciding if we like the way this is structured.

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+Configuration
+=============
+
+Configuration options
+
+.. automodule:: orcoursetrion.config
+
+.. autoattribute:: orcoursetrion.config.GH_API_KEY
+   :annotation: = GitHub API Key
```
Comment (Ferdi) [inline docs/api.rst]: Maybe "Github API token"  if this is what github calls  "Personal access tokens" ?

File: docs/command.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Command Line
+============
+
+There is an exposed command line interface that is available upon
+installation of the repository.  It can be run with ``orcoursetrion``, and
+``orcoursetrion --help`` will provide the most up to date help information.
+
+The command allows you to run commands that correspond to actions,
+currently the only supported action is ``create_export_repo``, and if
+your environment is setup correctly and you run ``orcoursetrion
```
Comment (Ferdi) [inline docs/command.rst]: How to know if the env is setup correctly ?  Can you add some info or link to docs ?

File: orcoursetrion/lib/github.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        """Initialize a requests session for use with this class by
+        specifying the base API endpoint and key.
+
+        Args:
+            api_url (str): Github API URL such as https://api.github.com/
+            api_key (str): Github API key for v3
+        """
+        self.api_url = api_url
+        self.session = requests.Session()
+        # Add API key to session headers
```
Comment (Ferdi) [inline orcoursetrion/lib/github.py]: Oauth token ?

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+setup(
+    name='orcoursetrion',
+    version=VERSION,
+    packages=find_packages(),
+    package_data={},
+    license='BSD Simplified',
+    author='Carson Gee',
+    author_email='x@carsongee.com',
+    url="http://github.com/carsongee/archelon",
```
Comment (Ferdi) [inline setup.py]: change it.

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    description=("Client connected to archelonc for Web shell history"),
```
Comment (Ferdi) [inline setup.py]: change description

**Comment by Ferdi** [comment]
Looks great. I had only a couple of nitpickings.

**Comment by carsongee** [comment]
I've addressed all the comments along with adding a quick start to the README

**Pull request merged**